dec-2019/Naloga8.java

```diff
@@ -0,0 +1,317 @@
+		
+		File vhod = new File(pot_vhod);
+		File izhod = new File(pot_izhod);
+		
+		Problem problem = preberiPodatke(vhod);
+		
+		int stNodov = stNodov(problem.dimenzija);
+		
+		PrintWriter p = null;
+		try {
+			p = new PrintWriter(new FileWriter(pot_izhod));
+		} catch (IOException e1) {
+
+			e1.printStackTrace();
+		}
+		
+		for(int i = 0; i< problem.stUkazov; i++) {
+			Node[] drevo = pretvori(problem, stNodov);
+			int[] test = poplavi(drevo, problem.ukazi[i], problem.dimenzija);
+			StringBuffer vrstica = print(test);
+			p.println(vrstica);
+		}
+		p.close();
+		
+	}
+	public static Problem preberiPodatke(File vhod) {
+		
+		Problem problem = new Problem();
+		Scanner scan = null;
+		try {
+			scan = new Scanner(vhod).useDelimiter("\\D");
+			
+		} catch (FileNotFoundException e) {
+
+			e.printStackTrace();
+		}
+		
+		int dimenzija = scan.nextInt();
+		scan.nextLine();
+		
+		problem.dimenzija = dimenzija;
+		
+		//zmeljevid visin
+		int[][] zemljevid = new int[dimenzija][dimenzija];
+		
+		for(int i = 0; i < dimenzija; i++) {
+			
+			String[] vrstica = scan.nextLine().split(",");
+			
+			for(int j = 0; j < dimenzija; j++) {
+				zemljevid[i][j] = Integer.parseInt(vrstica[j]);
+			}
+		}
+		problem.zemljevid = zemljevid;
+		
+		int stUkazov = scan.nextInt();
+		scan.nextLine();
+		problem.stUkazov = stUkazov;
+		
+		int[] ukazi = new int[stUkazov];
+		
+		for(int i = 0; i < stUkazov; i++) {
+			ukazi[i] = scan.nextInt();
+			scan.nextLine();
+		}
+		problem.ukazi = ukazi;
+		
+		scan.close();
+		
+		return problem;
+	}
+	//dobi inta, ki sta rezultat enega ukaza in zapise v file
+	public static StringBuffer print(int[] vrstica) {
+		
+		int stPop = vrstica[0];
+		int stPrebranih = vrstica[1];
+		StringBuffer rezultat = new StringBuffer();
+		
+		rezultat.append(stPop);
+		rezultat.append(",");
+		rezultat.append(stPrebranih);
+		
+		return rezultat;
+	}
+	//vzame zemljevid visin in vrne tabelo drevesa
+	public static Node[] pretvori(Problem p, int stNodov) {
+		Node[] drevo = new Node[stNodov];
+		int counter = stNodov-1;
+	
+		for(int i = counter; i >= 0; i--) {
+			drevo[i] = calculateMinMax(i, p, drevo);
+		}
+		return drevo;
+	}
+	public static int stNodov(int dimenzija) {
+		int stN = dimenzija*dimenzija;
+		int rez = stN;
+		while(stN > 1) {
+			stN = stN/4;
+			rez +=stN;
+		}
+		return rez;
+	}
+	//vrne rezultat po poplavljanju
+	public static int[] poplavi(Node[] tree, int visM, int dimenzija) {
+		Node[] drevo = tree;
+		int[] rez = new int[2];
+		int field = dimenzija*dimenzija;
+		int covCounter = 0; //steje kok mest pokrivajo presteti nodi
+		int floodCounter = 0;
+		int visitCounter = 0;
+		
+		for(int i = 0; i < drevo.length; i++) {
+			if(covCounter == field) {
+				rez[0] = floodCounter;
+				rez[1] = visitCounter;
+				return rez;
+			}
+			if(drevo[i] == null) {
+				continue;
+			}
+			if(drevo[i].min == drevo[i].max) {
+				deleteBelow(drevo, i);

```

Comment:
to bi moral narest že preden pokličeš poplavi - tole ni odvisno od višine vode. S tem bi bilo drevo, ki ga dobiš v funkcijo,  nespremenjeno. Java uči slabe navade - poplavi bi moral drevo jemat kot const.